compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -48,6 +49,25 @@
 
 namespace mlir::iree_compiler::AMDAIE {
 
+/// Command line option for selecting the target AIE device.
+static llvm::cl::opt<AMDAIEDevice> clAMDAIETargetDevice(
+    "iree-amdaie-target-device",
+    llvm::cl::desc("Sets the target device architecture."),
+    llvm::cl::values(
+        clEnumValN(AMDAIEDevice::xcvc1902, "xcvc1902", "The xcvc1902 device"),
+        clEnumValN(AMDAIEDevice::xcve2302, "xcve2302", "The xcve2302 device"),
+        clEnumValN(AMDAIEDevice::xcve2802, "xcve2802", "The xcve2802 device"),
+        clEnumValN(AMDAIEDevice::npu1, "npu1", "Default Phoenix NPU"),
+        clEnumValN(AMDAIEDevice::npu1_1col, "npu1_1col",
+                   "Phoenix NPU with a single column"),
+        clEnumValN(AMDAIEDevice::npu1_2col, "npu1_2col",
+                   "Phoenix NPU with two columns"),
+        clEnumValN(AMDAIEDevice::npu1_3col, "npu1_3col",
+                   "Phoenix NPU with three columns"),
+        clEnumValN(AMDAIEDevice::npu1_4col, "npu1_4col",
+                   "Phoenix NPU with four columns")),
+    llvm::cl::init(AMDAIEDevice::npu1_4col));
```

Comment:
I am wondering that shouldn't the codegen tiling strategy decide this rather than having a seperate target for each?

Reply:
Especially because we may be mixing and matching dispatches that are 1,2,4 columns?

Reply:
It's not just the number of columns, but also the physical device, for example strix would be `npu2_?` I believe. I guess the number of columns can be removed at some point.

Reply:
However, a bit more work is needed to decouple the device from the number of columns.

Reply:
The number of columns in that name should be considered legacy (from mlir-aie) and will be deprecated. For us (in-tree, today) those enum variants are only queried in a single place: https://github.com/nod-ai/iree-amd-aie/blob/main/runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.cc#L444-L448

```
    case AMDAIEDevice::npu1:
    case AMDAIEDevice::npu1_1col:
    case AMDAIEDevice::npu1_2col:
    case AMDAIEDevice::npu1_3col:
    case AMDAIEDevice::npu1_4col:
```

so the only bit of work left is to promote `partitionNumCols, partitionStartCol` to be the canonical attributes (rather than `device`).

Reply:
Ya npu2 as a target makes sense. I was only asking about the columns. Can codegen still make a 2 column design with the npu1_4col target? In that case this is okay.

Reply:
I didnt see Max's reply so in that case I am thinking partitionNumCols, partitionStartCol will be choosen by our codegen startegy so this makes sense thanks.